### PR TITLE
Fix sanitization in the contact form

### PR DESCRIPTION
**Pull request by gregpriday**
This resolves GH-68, GH-66 and GH-65.

### CODE REVIEW

File: widgets/contact/contact.php [hunk 1]
```diff
@@ -928,22 +928,33 @@ function spam_check( $post_vars, $email_fields, $instance ){
 	}
 
 	function send_mail( $email_fields, $instance ){
-		$body = '<strong>From:</strong> ' . $email_fields['name'] . ' <' . sanitize_email( $email_fields['email'] ) . ">\n\n";
+		$body = '<strong>From:</strong> <a href="mailto:' . sanitize_email( $email_fields['email'] ) . '">' . esc_html( $email_fields['name'] ) . "</a>>\n\n";
```
Comment (braamgenis) [inline widgets/contact/contact.php]: Looks like an extra '>' after the closing anchor tag.

File: widgets/contact/contact.php [hunk 2]
```diff
@@ -928,22 +928,33 @@ function spam_check( $post_vars, $email_fields, $instance ){
...
 
+	/**
+	 * Sanitize a value for an email header
+	 *
+	 * @param $value
+	 *
+	 * @return mixed
+	 */
+	static function sanitize_header( $value ){
+		return preg_replace('=((<CR>|<LF>|0x0A/%0A|0x0D/%0D|\\n|\\r)\S).*=i', null, $value);
```
Comment (braamgenis) [inline widgets/contact/contact.php]: Is this replacing any newline type character with a `null`? Why not just an empty string?

**Comment by braamgenis** [comment]
Couple of minor things, but otherwise good to merge when you're ready.

File: widgets/contact/contact.php (hunk 1 above)
Comment (gregpriday) [inline widgets/contact/contact.php]: Aaah, nicely spotted. Will fix that up.

File: widgets/contact/contact.php (hunk 2 above)
Comment (gregpriday) [inline widgets/contact/contact.php]: I think that's directly out of another (GPL compatible) library. I should probably add some form of attribution header to that function though.

**Pull request merged**